specs/lib/gltfPipelineSpec.js

```diff
@@ -0,0 +1,135 @@
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+    
+    it('optimizes a gltf JSON given a resourcePath', function() {
+        var gltf;
+        var gltfCopy;
+        var options = { 'resourcePath' : path.dirname(gltfPath) };
+        fs.readFile(gltfPath, function(err, data) {
+            if (err) {
+                throw err;
+            }
+            gltf = JSON.parse(data);
+            gltfCopy = clone(gltfCopy); 
+            addPipelineExtras(gltf);
+            
+            processJSON(gltf, options, function(gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+
+    it('optimizes a gltf JSON with extras', function() {
+        var gltfCopy;
+        var options = {};
+        readGltf(gltfPath, function(gltf) {
+            gltfCopy = clone(gltf);
+            processJSONWithExtras(gltf, options, function(gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+    
+    it('optimizes a binary glTF JSON with extras', function() {
+        var gltfCopy;
+        var options = {};
+        readGltf(glbPath, function(gltf) {
+            gltfCopy = clone(gltf);
+            processJSONWithExtras(gltf, options, function(gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+
+    it('optimizes a glTF file', function() {
+        var gltfCopy;
+        var options = {};
+        readGltf(gltfPath, function(gltf) {
+            gltfCopy = gltf;
+            processFile(gltfPath, options, function (gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+
+    it('optimizes a glb file', function() {
+        var gltfCopy;
+        var options = {};
+        readGltf(glbPath, function(gltf) {
+            gltfCopy = gltf;
+            processFile(glbPath, options, function (gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+
+    it('will write a file to the correct directory', function(done) {
+        var spy = spyOn(fs, 'writeFile').and.callFake(function(file, data, callback) {
+            callback();
+        });
+        var options = {};
+        processFileToDisk(gltfPath, outputPath, options, function() {
+            expect(spy).toHaveBeenCalled();
+            done();
+        });
+    });
+
+    it('will write a file', function(done) {
+        var spy = spyOn(fs, 'writeFile').and.callFake(function(file, data, callback) {
+            callback();
+        });
+        var options = {};
+        processFileToDisk(gltfPath, outputPath, options, function() {
+            expect(spy).toHaveBeenCalled();
+            done();
+        });
+    });
```

Comment:
I had an issue with trying to check for an expected output path. Because multiple files are written, it is difficult to test exactly what the output path would be (or at least I am not sure how I would go about it). 

For example: 
I tried `expect(spy).toHaveBeenCalledWith(outputPath, jasmine.any(String), jasmine.any(Function));` but the result ends up being

 `[ outputPath/output.gltf, <a stringified JSON blob>, <a function>], [ outputPath/output0VS.glsl, <a stringified JSON>, <a function>], ... `

Overall, I am just unsure how I can write a Jasmine expect for a spy that seems to see multiple outputs with different expected files. Maybe something with regular expressions?